src/components/Search/Search.tsx

```diff
@@ -22,20 +24,34 @@ const Search = ({ delayTimer = 500, hideBorder = false, name, onSearch, placehol
         };
     };
 
+    // eslint-disable-next-line react-hooks/exhaustive-deps
     const debouncedOnSearch = useCallback(debounce(onSearch, delayTimer), [onSearch]);
 
+    const handleInputChange = (event: React.FormEvent<HTMLFormElement>) => {
+        const value = (event.target as HTMLInputElement).value;
+        setSearchValue(value);
+        debouncedOnSearch(value);
+    };
+
+    const clearSearch = () => {
+        setSearchValue('');
+        debouncedOnSearch('');
+    };
+
     return (
-        <form
-            className='search'
-            onChange={event => debouncedOnSearch((event.target as HTMLInputElement).value)}
-            role='search'
-        >
+        <form className='search' onChange={event => handleInputChange(event)} role='search'>
             <Input
                 className={clsx({ 'border-none': hideBorder })}
                 label={placeholder}
                 leftPlaceholder={<LabelPairedSearchMdRegularIcon />}
                 name={name}
+                rightPlaceholder={
+                    searchValue && (
+                        <LegacyCloseCircle1pxIcon className='cursor-pointer' iconSize='xs' onClick={clearSearch} />
+                    )
+                }
                 type='search'
+                value={searchValue}
```

Comment:
can u please check if this is giving console warnings for giving the value without onchange handler where this search is used.